### DISCUSSION TITLE
BUG in "feat(http): Cleanup for HTTP cache #28381"

### DISCUSSION THREAD

**Discussion by mjnowen** [renovatebot/renovate#28439] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab v16.9.3, "renovateVersion": "37.301.0"

### Please tell us more about your question or problem

re: https://github.com/renovatebot/renovate/pull/28381

I believe this PR has introduced a bug. I have a Renovate MR in self-hosted GitLab with a >90 day http cache. Instead of it rotating (purging) my cache objects by default, it just prints a WARN message and returns. [cleanupHttpCache](https://github.com/renovatebot/renovate/blob/main/lib/util/cache/repository/http-cache.ts#L9)

I would expect >90 day http cache (by default=90) to not just return constantly with a WARN message but actually purge my object cache, but as the code is currently written, I never get to this line: https://github.com/renovatebot/renovate/blob/main/lib/util/cache/repository/http-cache.ts#L9

### Logs (if relevant)

<details><summary>Logs</summary>

```

$ renovate $RENOVATE_EXTRA_FLAGS
 INFO: Repository started (repository=<redacted>)
       "renovateVersion": "37.301.0"
 INFO: Branch created (repository=<redacted>, branch=renovate/configure)
       "commit": "<redacted>",
       "onboarding": true
 INFO: Dependency extraction complete (repository=<redacted>, baseBranch=main)
       "stats": {
         "managers": {
           "cloudbuild": {"fileCount": 1, "depCount": 1},
           "dockerfile": {"fileCount": 1, "depCount": 1},
           "gitlabci": {"fileCount": 1, "depCount": 47},
           "gomod": {"fileCount": 6, "depCount": 298},
           "terraform": {"fileCount": 10, "depCount": 28},
           "tflint-plugin": {"fileCount": 1, "depCount": 2}
         },
         "total": {"fileCount": 20, "depCount": [37](https://gitlab.<redacted>/-/jobs/6653718#L37)7}
       }
 INFO: Onboarding PR created (repository=<redacted>)
       "pr": "Pull Request #10"
 WARN: cleanupHttpCache: invalid cache data (repository=<redacted>)
 INFO: Repository finished (repository=<redacted>)
       "cloned": true,
       "durationMs": 28085
 INFO: Renovate was run at log level "info". Set LOG_LEVEL=debug in environment variables to see extended debug logs.
Saving cache for successful job

```

</details>

**Comment by rarkins** [comment]
Duplicate of #28432
